pyproject.toml

```diff
@@ -18,6 +18,18 @@ requires = [
 ]
 build-backend = "setuptools.build_meta"
 
+[tool.mypy]
+files = "trieste,tests,notebooks/**/*.py"
+ignore_missing_imports = true
+warn_redundant_casts = true
+warn_unused_ignores = true
+strict_equality = true
+
+[[tool.mypy.overrides]]
+module = ["trieste.*", "tests.*"]
```

Comment:
does this mean we need to specify types in tests? that seems unnecessary...

Reply:
it would mean that, yes. I think only the team can answer whether they want that or not

Reply:
i think this is generally the case with this PR. It does change the approach to contributions, so I expect everyone will want to know of this change

Reply:
I have raised a discussion point on this, I have mixed feelings about it, lets see

Reply:
I could omit `"tests.*"` for now. It would be very easy to add again later

Reply:
ok, lets do that for now and raise a separate PR later on if after a discussion we decide to go with that

Reply:
done